python/GafferUI/Button.py

```diff
@@ -84,6 +84,7 @@ def __init__( self, text="", image=None, hasFrame=True, highlightOnOver=True, **
 	def setHighlighted( self, highlighted ) :
 
 		GafferUI.Widget.setHighlighted( self, highlighted )
+		self.__highlightForHover = False
```

Comment:
Wasn't the entire point of `__highlightForHover` to disentangle the two sources of highlighting so they didn't fight each other? At the very least I think this needs an explanatory comment.

Reply:
Repro on `master`, choose any menu item from a button, the highlight state lingers. 
It was meant to de-couple it, but the issue is that https://github.com/GafferHQ/gaffer/blob/c0b4defec205a767913f0b8ea182ecc97fbc6631/python/GafferUI/MenuButton.py#L108 gets ignored as `self.__highlightForHover` is `True` from the mouse enter event. And so [this code](https://github.com/GafferHQ/gaffer/blob/c0b4defec205a767913f0b8ea182ecc97fbc6631/python/GafferUI/Button.py#L164), stops the button from being un-highlighted.

Reply:
Added comment [here]( ttps://github.com/GafferHQ/gaffer/pull/3842/commits/67f68f72fa9a4d7ebdbe6d60177b8e1bc7b15d9b)

Reply:
There is probably some better solution but it would maybe require friendship between the two so that `MenuButton` could clear `__highlightForHover` rather than using `setHighlighted`.

Reply:
I think that would be preferable. As it stands, this PR is breaking the well-defined behaviour of Buttons in order to support a (necessary) hack for MenuButtons. Since this is a hack, I would be happier to call `self.__Button_leave()` directly from MenuButton - it means we get to leave Buttons alone and isolate the hack to one spot. And the hack itself is more self explanatory because it calls the actual method that Qt has failed to call rather than a supposedly orthogonal method. We could make `_leave()` merely protected to facilitate this further, but since we don't intend anyone else to call it, and we know we're hacking, I think the bare-faced `__Button_leave()` approach is best.

Reply:
@johnhaddon Updated [here](https://github.com/GafferHQ/gaffer/pull/3842/commits/8e3e405ca69b526df0464a857db29827f8325b66).